### PR TITLE
f-restaurant-card@v0.12.0 - dish items

### PR DESCRIPTION
**Pull request by jamieomaguire**
v0.12.0
------------------------------
*December 15, 2021*

### Added
- Scrollable and stackable restaurant dish list component
- Dish list component unit tests
- Dish component unit tests
- Dish and Dish list storybook files

### Changed
- Updated restaurant dish styling
- Added dish state to restaurant card storybook

## Visual demos

Dec-15-2021 14-51-08
Screenshot 2021-12-15 at 14 52 17
Screenshot 2021-12-15 at 14 52 05
Screenshot 2021-12-15 at 14 49 47
Screenshot 2021-12-20 at 14 32 31

### CODE REVIEW

File: packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue [hunk 1]
```diff
@@ -246,6 +252,9 @@ export default {
             return this.deliveryTimeData.eta ||
                 this.deliveryTimeData.distance ||
                 this.deliveryTimeData.address;
+        },
+        hasDishes () {
+            return this.dishes?.length || false;
```
Comment (1itvinka) [inline packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue]:
```suggestion
            return !!this.dishes?.length;
```

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    </div>
+</template>
+
+<script>
+export default {
+    name: 'RestaurantDish',
+    props: {
+        /**
+         * Name of the dish
+         */
```
Comment (1itvinka) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue]: do we need this comment? the following two comments have some useful information about the format, but I wonder if we can also avoid them somehow too.

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDishes.vue [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    &:last-of-type {
+        margin-right: 0;
+        padding-right: $scrollOffset;
+        scroll-margin-right: $scrollOffset;
+        scroll-snap-align: center;
+    }
+
+    flex: 0 0 85%;
+    margin-bottom: 0;
+    margin-right: spacing();
```
Comment (1itvinka) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDishes.vue]: I would move this block before `return this.dishes?.length` block (after line 92)

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/_tests/RestaurantDish.test.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-unused-vars */
+import { mount } from '@vue/test-utils';
+import sut from '../RestaurantDish.vue';
+
```
Comment (AlixRichards) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/_tests/RestaurantDish.test.js:4]: What does `sut` mean/short for?

**Review by AlixRichards — COMMENTED** [review]
Looks good! I have a question about the layout on the `desktop` view. To me the `new` and ratings section look like they are missing a little margin between them. I haven't seen the designs so this could be intentional. However I think it would be worth raising as it will look more consistent between screensizes

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue (hunk 2 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue]: happy to remove! it does speak for itself - you're right

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/_tests/RestaurantDish.test.js (hunk 4 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/_tests/RestaurantDish.test.js:4]: `system under test` - it's a relatively common shorthand for 'the thing being tested' in unit tests

**Comment by jamieomaguire** [comment]
> Looks good! I have a question about the layout on the `desktop` view. To me the `new` and ratings section look like they are missing a little margin between them. I haven't seen the designs so this could be intentional. However I think it would be worth raising as it will look more consistent between screensizes

Cheers Alix! All the items within the restaurant card have not being laid out properly yet. The focus right now is to get all the data points styled, with appropriate render logic. Then once everything exists within the card to support all variations, I'm going to add the layout styling. That's when the `new` and ratings section will have correct spacing etc. :)

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDishes.vue (hunk 3 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDishes.vue]: shall do! good spot

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/_tests/RestaurantDish.test.js (hunk 4 above)
Comment (AlixRichards) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/_tests/RestaurantDish.test.js:4]: Cool 👍

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        portion: {
+            type: String,
+            default: null
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+.c-restaurantCard-dish {
```
Comment (AlixRichards) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue:68]: nit: indentation in this declaration doesn't match the rest below

Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue:68]: oops!

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue [hunk 6]
```diff
@@ -0,0 +1,124 @@
+<template>
+    <div :class="[$style['c-restaurantCard-dish']]">
+        <p :class="[$style['c-restaurantCard-dish-description']]">
+            <span
+                data-test-id="restaurant-dish-name"
+                :class="[$style['c-restaurantCard-dish-name']]">
+                {{ name }}
+            </span>
+            <span
+                v-if="calories"
```
Comment (1itvinka) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue:10]:
```suggestion
                v-if="calories || portion"
```

**Review by AlixRichards — APPROVED** [review]

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                :class="[$style['c-restaurantCard-dish-nutritionalInfo']]">
+                <span
+                    data-test-id="restaurant-dish-calories"
```
Comment (1itvinka) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue:13]:
```suggestion
                    v-if="calories"
                    data-test-id="restaurant-dish-calories"
```

not sure about it but it seems like even if there's portion info and no calories info we don't show this info at all. not sure it it should work this way or not

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue (hunk 6 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue:10]: I'm unsure about this. I don't think portion makes sense without calories in this context, but I could be mistaken. Considering this is a future requirement I'd prefer to leave as is for now and fix forward once we have done more exploration around the calories requirement next year :) I'll revisit this logic then.

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue (hunk 7 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue:13]: @1itvinka to echo my reply to the other comment - I think for now I am going to leave as is. To my mind, it only makes sense to show portions if there are calories. So if there's no reason for the extra calories check here.

As the requirements haven't been fully explored yet, I'd like to leave this render logic as is for now with the plan to revisit next year once we have a better understanding of the requirements. This is essentially just a starting point for that work :)

Comment (1itvinka) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantDishes/RestaurantDish.vue:13]: Yeah, let's see how it goes. Personally, I don't care about calories, but it's good to know the portion size 😂

**Review by 1itvinka — APPROVED** [review]

**Pull request merged**